### ISSUE TITLE
Dashboard OAuth Client not deleted when deleting Service Broker

### ISSUE THREAD

**Issue by bkrannich** [cloudfoundry/cloud_controller_ng#1032]
## Issue

When registering a service broker in Cloud Foundry, you can specify an OAuth Client/Secret in the Service Catalog which is then created during service broker registration and used for the Dashboard SSO handling.

However, when deleting the service broker registration the corresponding OAuth Client is not deleted. The question is whether this is the expected behavior or a bug?

## Context

See https://docs.cloudfoundry.org/services/dashboard-sso.html#registering-dashboard-client

## Steps to Reproduce

Register a service broker declaring a ```dashboard_client```.

## Expected result

One could think that it is the right thing to do to also again delete the ```dashboard_client``` created.

## Current result

The ```dashboard_client``` stays in the system even after the service broker has been unregistered.

## Possible Fix

Delete the ```dashboard_client``` when de-registering the service broker. Take care to delete the right client and prevent the broker from specifying a completely different client in it's catalog which is then delete instead of the originally created one.

**Comment by anniesing** [comment]
Hi @bkrannich,

Thanks for submitting. This looks like a question for @zrob, so I'm labeling this as a PM Review issue.

Thanks,
Annie

**Comment by lisamcho** [comment]
@bkrannich can you provide additional reproduction steps? In particular, where in the system do you see the `dashboard_client` after you have deleted the service broker?

I could not reproduce this bug with the steps below:

1. Create the broker**:
Our data.json file lists a known dashboard_client
```
...
            "dashboard_client": {
              "id": "dashboard-client-id",
              "secret": "dashboard-client-secret",
              "redirect_uri": "<redirect uri>"
            },
...
```

`cf push broker`
`cf create-service-broker test-broker user password  <broker url>`
`cf enable-service-access <service name>

2. Verify that the uaac client 'dashboard-client-id' is listed with the uaac cli:
`uaac clients`

3. Delete the service broker
`cf delete-service-broker test-broker`

4. Verify that the uaac client 'dashboard-client-id' is no longer listed with the uaac cli
`uaac clients`

**Dev Note: We can easily push this broker by modifying our `setup_new_broker.rb` script here with a known dashboard client id:
https://github.com/cloudfoundry/cf-acceptance-tests/blob/6768f8d4d43908468f3bb53687bad85f2027ef6d/assets/service_broker/setup_new_broker.rb#L37-L38

**Comment by bkrannich** [comment]
Hi @lisamcho,

Thank you very much for looking into the topic! With your finding of you not being able to reproduce the problem I went back to the colleague who brought up the topic on our end. I asked him to check if the issue persists with the latest cf-deployment we are using meanwhile. Indeed, he confirmed that he also cannot reproduce the issue anymore in our environment.

I think we could go back root-causing if this was an issue in some old cf-release/cf-deployment and what was changed so that the problem no longer exists. But as we already created effort for you (very sorry about that) and the result of such an investigation wouldn't impact the currently available cf-deployment I would suggest closing the issue.

Thanks again,
Bernd
